Source/Core/Common/HttpRequest.h

```diff
@@ -18,7 +18,8 @@ namespace Common
 class HttpRequest final
 {
 public:
-  explicit HttpRequest(std::chrono::milliseconds timeout_ms = std::chrono::milliseconds{3000});
+  explicit HttpRequest(std::chrono::milliseconds timeout_ms = std::chrono::milliseconds{3000},
+                       void* callback = nullptr);
```

Comment:
This should probably be named `progress_callback` rather than the ambiguous `callback`

Reply:
Also it should be wrapped in such a way that we don't expose CURL internals through the public API. e.g. define our own callback type and have something that does the translation in Impl.